tests/extmod/vfs_posix_ilistdir_filter.py

```diff
@@ -0,0 +1,45 @@
+# Test ilistdir filter of . and .. for VfsPosix.
+
+try:
+    import os
+
+    os.VfsPosix
+except (ImportError, AttributeError):
+    print("SKIP")
+    raise SystemExit
+
+
+def test(testdir):
+    vfs = os.VfsPosix(testdir)
+
+    expected = {".a": True, "..a": True, "...a": True, "a.": True}
+    actual = {}
+
+    for dir in expected:
+        vfs.mkdir(dir)
+
+    for entry in vfs.ilistdir("/"):
```

Comment:
Since the windows test is failing, perhaps this needs to be `op.path.sep` instead of `"/"`?

Reply:
Perhaps.  I was trying to follow the pattern I saw in vfs_posix_ilistdir_del.py and it also does vfs.ilistdir("/").  But I guess if I take your advice above about printing and comparing to the .exp file, it may be more clear why it is failing on Windows.

Reply:
Interesting... Windows strips the trailing `.` from `'a.'`.

https://stackoverflow.com/questions/11681207/how-to-create-a-filename-with-a-trailing-period-in-windows

Reply:
It is now clear why the test is failing under Windows.  The test is creating a directory called 'a.' because I am trying to give a series of inputs with periods in different locations to ensure they aren't picked up by the filter in ilistdir().  But on Windows, a file/directory called 'a.' is actually just 'a'.  This dates right back to the 8.3 naming scheme from DOS I think.

Here is what the test output is saying the difference is:

```
-['...a', '..a', '.a', 'a.']
+['...a', '..a', '.a', 'a']
```

Note that 'a.' from the expected is just 'a' in the actual output under Windows.  I don't really need 'a.' so maybe I will remove it from the test but it is annoying that I can't cover this case because of a weird Windows behaviour.  At least that is what I think is happening.

Reply:
Is it possible to add `'a.b'` and `'a..'` and/or `'a..b`' to the list?

Reply:
I am skeptical that 'a..' would work on Windows because it still ends in a dot and it does seem like that will be a problem on that platform.  But I have added the other two to the test.

Reply:
Great, thanks!